src/main/scala/com/databricks/spark/xml/XmlInputFormat.scala

```diff
@@ -189,9 +192,23 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
     false
   }
 
-  private def checkEmptyTag(currentLetter: Int, position: Int): Boolean = {
+  private def checkEmptyTag(currentLetter: Int, position: Int,
+                            buffer: DataOutputBuffer): Boolean = {
+    def checkStartTagBefore = {
+      val buf = Seq('<'.toByte)
+      val rootTagName = buffer.getData
```

Comment:
Why define a method here?
I don't think this is efficient enough as it makes a few copies of much of the buffer.

Reply:
Because it's used only here.

> I don't think this is efficient enough as it makes a few copies of much of the buffer.

maybe, but this code run only in rare situations
code in buffet not so easy to read